plugins/repository-azure/build.gradle

```diff
@@ -56,6 +56,23 @@ dependencies {
   api "io.netty:netty-transport-native-unix-common:${versions.netty}"
   implementation project(':modules:transport-netty4')
   api 'com.azure:azure-storage-blob:12.23.0'
+  api 'com.azure:azure-identity:1.11.4'
+  // Start of transitive dependencies for azure-identity
+  api 'com.microsoft.azure:msal4j-persistence-extension:1.2.0'
+  // Both azure-identity:1.11.4 and msal4j-persistence-extension:1.2.0 has compile dependency on different versions of jna-platform,
+  // selected the higher version which is 5.13.0
+  api 'net.java.dev.jna:jna-platform:5.13.0'
```

Comment:
@chengwushi-netapp could you please explain why these dependencies are needed? I don't see them here [1] and [2].

[1] https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/identity/azure-identity/README.md
[2] https://mvnrepository.com/artifact/com.azure/azure-identity/1.11.4

Reply:
Hello @reta, please see attached the design doc i shared earlier, the section `Rationale Behind Adding Dependencies` in the doc should explain why the transitive dependencies `com.microsoft.azure:msal4j`, `com.nimbusds:oauth2-oidc-sdk`, and `net.minidev:json-smart` are added.  

All the other transitive dependencies were added to pass the `thirdPartyAudit` gradle check. 

[Managed-Identity-Support-Design-Doc-GitHub.docx](https://github.com/opensearch-project/OpenSearch/files/14953618/Managed-Identity-Support-Design-Doc-GitHub.docx)